File: OCaml/lib/executable.ml
open Question

module Executable = functor (M: Question) -> struct
  let execute = function
    1 -> M.question_1 ()
    | 2 -> M.question_2 ()
    | 3 -> M.question_3 ()
    | 4 -> M.question_4 ()
    | _ -> ()
end

File: FSharp/HelloWorld/Lib/Executable.fs
namespace HelloWorld.Lib

type Executable(m: IQuestion) =
    member public this.Execute num =
        match num with
        | 1 -> m.Question1 ()
        | 2 -> m.Question2 ()
        | 3 -> m.Question3 ()
        | 4 -> m.Question4 ()
        | _ -> ()